### PR TITLE
Add default flags for wasm threading

### PR DESCRIPTION
**Pull request by Lambourl**

### CODE REVIEW

File: wasm.cmake [hunk 1]
```diff
@@ -259,6 +259,7 @@ set(CMAKE_C_FLAGS_RELWITHDEBINFO "-O2" CACHE STRING "Emscripten-overridden CMAKE
 set(CMAKE_CXX_FLAGS_RELEASE "-DNDEBUG -O2" CACHE STRING "Emscripten-overridden CMAKE_CXX_FLAGS_RELEASE")
 set(CMAKE_CXX_FLAGS_MINSIZEREL "-DNDEBUG -Os" CACHE STRING "Emscripten-overridden CMAKE_CXX_FLAGS_MINSIZEREL")
 set(CMAKE_CXX_FLAGS_RELWITHDEBINFO "-O2" CACHE STRING "Emscripten-overridden CMAKE_CXX_FLAGS_RELWITHDEBINFO")
+set (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -s USE_PTHREADS=1")
```
Comment (daminetreg) [inline wasm.cmake]: I realize we need thems in the CMAKE_C_FLAGS too and  please add them in the CACHE as the other variable around. Toolchain file is parsed early, by adding variables to the cache you ensure they get treated correctly on future call to cmake.

```cmake
set (CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -Wl,--shared-memory,--no-check-features --bind -s NO_EXIT_RUNTIME=1" CACHE STRING "Emscripten-overridden CMAKE_EXE_LINKER_FLAGS")
set (CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} --bind -s NO_EXIT_RUNTIME=1" CACHE STRING "Emscripten-overridden")
```

**Review by daminetreg — CHANGES_REQUESTED** [review]
Thanks for the change, the issue wasn't descriptive enough about the work to be done, can you perform the change asked and therefore fix nxxm/nxxm#78 accordingly.

I pushed a test in nxxm/nxxm#78 that break when it's not correct.

File: wasm.cmake (hunk 1 above)
Comment (daminetreg) [reply wasm.cmake]: Additionally for sourcemaps to be generated in debug mode. We need to add -g4 to CMAKE_CXX_FLAGS_DEBUG.

File: wasm.cmake [hunk 2]
```diff
@@ -259,7 +259,7 @@ set(CMAKE_C_FLAGS_RELWITHDEBINFO "-O2" CACHE STRING "Emscripten-overridden CMAKE
 set(CMAKE_CXX_FLAGS_RELEASE "-DNDEBUG -O2" CACHE STRING "Emscripten-overridden CMAKE_CXX_FLAGS_RELEASE")
 set(CMAKE_CXX_FLAGS_MINSIZEREL "-DNDEBUG -Os" CACHE STRING "Emscripten-overridden CMAKE_CXX_FLAGS_MINSIZEREL")
 set(CMAKE_CXX_FLAGS_RELWITHDEBINFO "-O2" CACHE STRING "Emscripten-overridden CMAKE_CXX_FLAGS_RELWITHDEBINFO")
-set (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -s USE_PTHREADS=1")
+set (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS}  -s CACHE USE_PTHREADS=1")
```
Comment (daminetreg) [inline wasm.cmake]: The cache keyword should be after the string actually.

**Comment by daminetreg** [comment]
Thanks 😊 Don’t forget the CMAKE_C_FLAGS too.

File: wasm.cmake (hunk 1 above)
Comment (Lambourl) [reply wasm.cmake]: CMAKE_CXX_FLAGS_DEBUG doesn't exist. Does I create it ?

Comment (daminetreg) [reply wasm.cmake]: Yes please for -g4

File: wasm.cmake [hunk 3]
```diff
@@ -256,10 +256,11 @@ set(CMAKE_DL_LIBS)
 set(CMAKE_C_FLAGS_RELEASE "-DNDEBUG -O2" CACHE STRING "Emscripten-overridden CMAKE_C_FLAGS_RELEASE")
 set(CMAKE_C_FLAGS_MINSIZEREL "-DNDEBUG -Os" CACHE STRING "Emscripten-overridden CMAKE_C_FLAGS_MINSIZEREL")
 set(CMAKE_C_FLAGS_RELWITHDEBINFO "-O2" CACHE STRING "Emscripten-overridden CMAKE_C_FLAGS_RELWITHDEBINFO")
+set (CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -s USE_PTHREADS=1" CACHE STRIG "")
 set(CMAKE_CXX_FLAGS_RELEASE "-DNDEBUG -O2" CACHE STRING "Emscripten-overridden CMAKE_CXX_FLAGS_RELEASE")
 set(CMAKE_CXX_FLAGS_MINSIZEREL "-DNDEBUG -Os" CACHE STRING "Emscripten-overridden CMAKE_CXX_FLAGS_MINSIZEREL")
 set(CMAKE_CXX_FLAGS_RELWITHDEBINFO "-O2" CACHE STRING "Emscripten-overridden CMAKE_CXX_FLAGS_RELWITHDEBINFO")
-set (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -s USE_PTHREADS=1")
+set (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -s USE_PTHREADS=1" CACHE STRIG "")
```
Comment (daminetreg) [inline wasm.cmake]: STRIG != STRING

**Review by daminetreg — APPROVED** [review]

**Pull request merged**